### ISSUE TITLE
constraint errors when changing layout

### ISSUE THREAD

**Issue by kiwo12345** [Yalantis/DisplaySwitcher#20]
**report a bug**

When testing the demo I sometimes get a contraint error when changing layout:
```

2017-01-23 02:15:58.926309 DisplaySwitcher_Example[8893:1432862] [LayoutConstraints] Unable to simultaneously satisfy constraints.
	Probably at least one of the constraints in the following list is one you don't want. 
	Try this: 
		(1) look at each constraint and try to figure out which you don't expect; 
		(2) find the code that added the unwanted constraint or constraints and fix it. 
(
    "<NSLayoutConstraint:0x61800008c170 H:|-(10)-[UILabel:0x7fe945b1ded0'Marry Kennedy']   (active, names: '|':UIView:0x7fe945b1dd30 )>",
    "<NSLayoutConstraint:0x61800008b270 H:[UILabel:0x7fe945b1ded0'Marry Kennedy']-(11)-|   (active, names: '|':UIView:0x7fe945b1dd30 )>",
    "<NSLayoutConstraint:0x61800008b6d0 UIImageView:0x7fe945b1e160.width == 1   (active)>",
    "<NSLayoutConstraint:0x61800008c3f0 UIImageView:0x7fe945b1e160.width == UIView:0x7fe945b1dd30.width   (active)>"
)

Will attempt to recover by breaking constraint 
<NSLayoutConstraint:0x61800008b270 H:[UILabel:0x7fe945b1ded0'Marry Kennedy']-(11)-|   (active, names: '|':UIView:0x7fe945b1dd30 )>

Make a symbolic breakpoint at UIViewAlertForUnsatisfiableConstraints to catch this in the debugger.
The methods in the UIConstraintBasedLayoutDebugging category on UIView listed in <UIKit/UIView.h> may also be helpful.
Hi 0
Hi 0
Hi 0
Hi 0
Hi 0
2017-01-23 02:16:17.912348 DisplaySwitcher_Example[8893:1432862] [MC] System group container for systemgroup.com.apple.configurationprofiles path is /Users/Kevin/Library/Developer/CoreSimulator/Devices/28151775-0F99-4CC1-9CF3-3060CC67F995/data/Containers/Shared/SystemGroup/systemgroup.com.apple.configurationprofiles
2017-01-23 02:16:17.912974 DisplaySwitcher_Example[8893:1432862] [MC] Reading from private effective user settings.
2017-01-23 02:16:31.670696 DisplaySwitcher_Example[8893:1432862] [LayoutConstraints] Unable to simultaneously satisfy constraints.
	Probably at least one of the constraints in the following list is one you don't want. 
	Try this: 
		(1) look at each constraint and try to figure out which you don't expect; 
		(2) find the code that added the unwanted constraint or constraints and fix it. 
(
    "<NSLayoutConstraint:0x608000084e70 H:|-(10)-[UILabel:0x7fe942437b70'Leo Nicholson']   (active, names: '|':UIView:0x7fe9424379d0 )>",
    "<NSLayoutConstraint:0x60800008acd0 H:[UILabel:0x7fe942437b70'Leo Nicholson']-(11)-|   (active, names: '|':UIView:0x7fe9424379d0 )>",
    "<NSLayoutConstraint:0x608000094730 UIImageView:0x7fe942437e00.width == 3   (active)>",
    "<NSLayoutConstraint:0x608000092a20 UIImageView:0x7fe942437e00.width == UIView:0x7fe9424379d0.width   (active)>"
)

Will attempt to recover by breaking constraint 
<NSLayoutConstraint:0x60800008acd0 H:[UILabel:0x7fe942437b70'Leo Nicholson']-(11)-|   (active, names: '|':UIView:0x7fe9424379d0 )>

Make a symbolic breakpoint at UIViewAlertForUnsatisfiableConstraints to catch this in the debugger.
The methods in the UIConstraintBasedLayoutDebugging category on UIView listed in <UIKit/UIView.h> may also be helpful.
2017-01-23 02:16:38.154543 DisplaySwitcher_Example[8893:1432862] [LayoutConstraints] Unable to simultaneously satisfy constraints.
	Probably at least one of the constraints in the following list is one you don't want. 
	Try this: 
		(1) look at each constraint and try to figure out which you don't expect; 
		(2) find the code that added the unwanted constraint or constraints and fix it. 
(
    "<NSLayoutConstraint:0x61800008ff50 H:|-(10)-[UILabel:0x7fe942700440'Monica Lamberts']   (active, names: '|':UIView:0x7fe942702c80 )>",
    "<NSLayoutConstraint:0x618000088390 H:[UILabel:0x7fe942700440'Monica Lamberts']-(11)-|   (active, names: '|':UIView:0x7fe942702c80 )>",
    "<NSLayoutConstraint:0x618000082710 UIImageView:0x7fe9427013f0.width == 3   (active)>",
    "<NSLayoutConstraint:0x618000084060 UIImageView:0x7fe9427013f0.width == UIView:0x7fe942702c80.width   (active)>"
)

Will attempt to recover by breaking constraint 
<NSLayoutConstraint:0x618000088390 H:[UILabel:0x7fe942700440'Monica Lamberts']-(11)-|   (active, names: '|':UIView:0x7fe942702c80 )>

Make a symbolic breakpoint at UIViewAlertForUnsatisfiableConstraints to catch this in the debugger.
The methods in the UIConstraintBasedLayoutDebugging category on UIView listed in <UIKit/UIView.h> may also be helpful.
```

I am running Xcode 8.2.1
Simulator: Iphone 5s ios 10.2

**Comment by kiwo12345** [comment]
2017-01-23 02:38:10.850035 DisplaySwitcher_Example[9356:1465466] [LayoutConstraints] Unable to simultaneously satisfy constraints.
	Probably at least one of the constraints in the following list is one you don't want. 
	Try this: 
		(1) look at each constraint and try to figure out which you don't expect; 
		(2) find the code that added the unwanted constraint or constraints and fix it. 
(
    "<NSLayoutConstraint:0x610000099aa0 H:|-(10)-[UILabel:0x7fa00ff487e0'Jack Hoggarth']   (active, names: '|':UIView:0x7fa00ff48640 )>",
    "<NSLayoutConstraint:0x61000009a4f0 H:[UILabel:0x7fa00ff487e0'Jack Hoggarth']-(11)-|   (active, names: '|':UIView:0x7fa00ff48640 )>",
    "<NSLayoutConstraint:0x61000008db60 UIImageView:0x7fa00ff48a70.width == 1   (active)>",
    "<NSLayoutConstraint:0x610000097520 UIImageView:0x7fa00ff48a70.width == UIView:0x7fa00ff48640.width   (active)>"
)

Will attempt to recover by breaking constraint 
<NSLayoutConstraint:0x61000009a4f0 H:[UILabel:0x7fa00ff487e0'Jack Hoggarth']-(11)-|   (active, names: '|':UIView:0x7fa00ff48640 )>

Make a symbolic breakpoint at UIViewAlertForUnsatisfiableConstraints to catch this in the debugger.
The methods in the UIConstraintBasedLayoutDebugging category on UIView listed in <UIKit/UIView.h> may also be helpful.
2017-01-23 02:38:10.851424 DisplaySwitcher_Example[9356:1465466] [LayoutConstraints] Unable to simultaneously satisfy constraints.
	Probably at least one of the constraints in the following list is one you don't want. 
	Try this: 
		(1) look at each constraint and try to figure out which you don't expect; 
		(2) find the code that added the unwanted constraint or constraints and fix it. 
(
    "<NSLayoutConstraint:0x610000099aa0 H:|-(10)-[UILabel:0x7fa00ff487e0'Jack Hoggarth']   (active, names: '|':UIView:0x7fa00ff48640 )>",
    "<NSLayoutConstraint:0x61000009a4f0 H:[UILabel:0x7fa00ff487e0'Jack Hoggarth']-(11)-|   (active, names: '|':UIView:0x7fa00ff48640 )>",
    "<NSLayoutConstraint:0x61000008db60 UIImageView:0x7fa00ff48a70.width == 1   (active)>",
    "<NSLayoutConstraint:0x610000097520 UIImageView:0x7fa00ff48a70.width == UIView:0x7fa00ff48640.width   (active)>"
)

Will attempt to recover by breaking constraint 
<NSLayoutConstraint:0x61000009a4f0 H:[UILabel:0x7fa00ff487e0'Jack Hoggarth']-(11)-|   (active, names: '|':UIView:0x7fa00ff48640 )>

Make a symbolic breakpoint at UIViewAlertForUnsatisfiableConstraints to catch this in the debugger.
The methods in the UIConstraintBasedLayoutDebugging category on UIView listed in <UIKit/UIView.h> may also be helpful.
2017-01-23 02:38:10.853298 DisplaySwitcher_Example[9356:1465466] [LayoutConstraints] Unable to simultaneously satisfy constraints.
	Probably at least one of the constraints in the following list is one you don't want. 
	Try this: 
		(1) look at each constraint and try to figure out which you don't expect; 
		(2) find the code that added the unwanted constraint or constraints and fix it. 
(
    "<NSLayoutConstraint:0x610000094280 H:|-(10)-[UILabel:0x7fa00ff36210'Tom Jerome']   (active, names: '|':UIView:0x7fa00ff36070 )>",
    "<NSLayoutConstraint:0x6100000908b0 H:[UILabel:0x7fa00ff36210'Tom Jerome']-(11)-|   (active, names: '|':UIView:0x7fa00ff36070 )>",
    "<NSLayoutConstraint:0x61000009a5e0 UIImageView:0x7fa00ff18eb0.width == 1   (active)>",
    "<NSLayoutConstraint:0x6100000940f0 UIImageView:0x7fa00ff18eb0.width == UIView:0x7fa00ff36070.width   (active)>"
)

Will attempt to recover by breaking constraint 
<NSLayoutConstraint:0x6100000908b0 H:[UILabel:0x7fa00ff36210'Tom Jerome']-(11)-|   (active, names: '|':UIView:0x7fa00ff36070 )>

Make a symbolic breakpoint at UIViewAlertForUnsatisfiableConstraints to catch this in the debugger.
The methods in the UIConstraintBasedLayoutDebugging category on UIView listed in <UIKit/UIView.h> may also be helpful.
2017-01-23 02:38:10.854409 DisplaySwitcher_Example[9356:1465466] [LayoutConstraints] Unable to simultaneously satisfy constraints.
	Probably at least one of the constraints in the following list is one you don't want. 
	Try this: 
		(1) look at each constraint and try to figure out which you don't expect; 
		(2) find the code that added the unwanted constraint or constraints and fix it. 
(
    "<NSLayoutConstraint:0x610000094280 H:|-(10)-[UILabel:0x7fa00ff36210'Tom Jerome']   (active, names: '|':UIView:0x7fa00ff36070 )>",
    "<NSLayoutConstraint:0x6100000908b0 H:[UILabel:0x7fa00ff36210'Tom Jerome']-(11)-|   (active, names: '|':UIView:0x7fa00ff36070 )>",
    "<NSLayoutConstraint:0x61000009a5e0 UIImageView:0x7fa00ff18eb0.width == 1   (active)>",
    "<NSLayoutConstraint:0x6100000940f0 UIImageView:0x7fa00ff18eb0.width == UIView:0x7fa00ff36070.width   (active)>"
)

Will attempt to recover by breaking constraint 
<NSLayoutConstraint:0x6100000908b0 H:[UILabel:0x7fa00ff36210'Tom Jerome']-(11)-|   (active, names: '|':UIView:0x7fa00ff36070 )>

Make a symbolic breakpoint at UIViewAlertForUnsatisfiableConstraints to catch this in the debugger.
The methods in the UIConstraintBasedLayoutDebugging category on UIView listed in <UIKit/UIView.h> may also be helpful.
2017-01-23 02:38:10.855425 DisplaySwitcher_Example[9356:1465466] [LayoutConstraints] Unable to simultaneously satisfy constraints.
	Probably at least one of the constraints in the following list is one you don't want. 
	Try this: 
		(1) look at each constraint and try to figure out which you don't expect; 
		(2) find the code that added the unwanted constraint or constraints and fix it. 
(
    "<NSLayoutConstraint:0x618000090ef0 H:|-(10)-[UILabel:0x7fa01200b390'Leo Nicholson']   (active, names: '|':UIView:0x7fa012015410 )>",
    "<NSLayoutConstraint:0x618000092750 H:[UILabel:0x7fa01200b390'Leo Nicholson']-(11)-|   (active, names: '|':UIView:0x7fa012015410 )>",
    "<NSLayoutConstraint:0x618000090fe0 UIImageView:0x7fa01201c080.width == 1   (active)>",
    "<NSLayoutConstraint:0x618000094550 UIImageView:0x7fa01201c080.width == UIView:0x7fa012015410.width   (active)>"
)

Will attempt to recover by breaking constraint 
<NSLayoutConstraint:0x618000092750 H:[UILabel:0x7fa01200b390'Leo Nicholson']-(11)-|   (active, names: '|':UIView:0x7fa012015410 )>

Make a symbolic breakpoint at UIViewAlertForUnsatisfiableConstraints to catch this in the debugger.
The methods in the UIConstraintBasedLayoutDebugging category on UIView listed in <UIKit/UIView.h> may also be helpful.
2017-01-23 02:38:10.856374 DisplaySwitcher_Example[9356:1465466] [LayoutConstraints] Unable to simultaneously satisfy constraints.
	Probably at least one of the constraints in the following list is one you don't want. 
	Try this: 
		(1) look at each constraint and try to figure out which you don't expect; 
		(2) find the code that added the unwanted constraint or constraints and fix it. 
(
    "<NSLayoutConstraint:0x618000090ef0 H:|-(10)-[UILabel:0x7fa01200b390'Leo Nicholson']   (active, names: '|':UIView:0x7fa012015410 )>",
    "<NSLayoutConstraint:0x618000092750 H:[UILabel:0x7fa01200b390'Leo Nicholson']-(11)-|   (active, names: '|':UIView:0x7fa012015410 )>",
    "<NSLayoutConstraint:0x618000090fe0 UIImageView:0x7fa01201c080.width == 1   (active)>",
    "<NSLayoutConstraint:0x618000094550 UIImageView:0x7fa01201c080.width == UIView:0x7fa012015410.width   (active)>"
)

Will attempt to recover by breaking constraint 
<NSLayoutConstraint:0x618000092750 H:[UILabel:0x7fa01200b390'Leo Nicholson']-(11)-|   (active, names: '|':UIView:0x7fa012015410 )>

Make a symbolic breakpoint at UIViewAlertForUnsatisfiableConstraints to catch this in the debugger.
The methods in the UIConstraintBasedLayoutDebugging category on UIView listed in <UIKit/UIView.h> may also be helpful.
2017-01-23 02:38:10.857392 DisplaySwitcher_Example[9356:1465466] [LayoutConstraints] Unable to simultaneously satisfy constraints.
	Probably at least one of the constraints in the following list is one you don't want. 
	Try this: 
		(1) look at each constraint and try to figure out which you don't expect; 
		(2) find the code that added the unwanted constraint or constraints and fix it. 
(
    "<NSLayoutConstraint:0x618000094be0 H:|-(10)-[UILabel:0x7fa01202bbf0'Grase Kendal']   (active, names: '|':UIView:0x7fa012024640 )>",
    "<NSLayoutConstraint:0x618000096170 H:[UILabel:0x7fa01202bbf0'Grase Kendal']-(11)-|   (active, names: '|':UIView:0x7fa012024640 )>",
    "<NSLayoutConstraint:0x6180000907c0 UIImageView:0x7fa012016350.width == 1   (active)>",
    "<NSLayoutConstraint:0x61800008dcf0 UIImageView:0x7fa012016350.width == UIView:0x7fa012024640.width   (active)>"
)

Will attempt to recover by breaking constraint 
<NSLayoutConstraint:0x618000096170 H:[UILabel:0x7fa01202bbf0'Grase Kendal']-(11)-|   (active, names: '|':UIView:0x7fa012024640 )>

Make a symbolic breakpoint at UIViewAlertForUnsatisfiableConstraints to catch this in the debugger.
The methods in the UIConstraintBasedLayoutDebugging category on UIView listed in <UIKit/UIView.h> may also be helpful.
2017-01-23 02:38:10.858465 DisplaySwitcher_Example[9356:1465466] [LayoutConstraints] Unable to simultaneously satisfy constraints.
	Probably at least one of the constraints in the following list is one you don't want. 
	Try this: 
		(1) look at each constraint and try to figure out which you don't expect; 
		(2) find the code that added the unwanted constraint or constraints and fix it. 
(
    "<NSLayoutConstraint:0x618000094be0 H:|-(10)-[UILabel:0x7fa01202bbf0'Grase Kendal']   (active, names: '|':UIView:0x7fa012024640 )>",
    "<NSLayoutConstraint:0x618000096170 H:[UILabel:0x7fa01202bbf0'Grase Kendal']-(11)-|   (active, names: '|':UIView:0x7fa012024640 )>",
    "<NSLayoutConstraint:0x6180000907c0 UIImageView:0x7fa012016350.width == 1   (active)>",
    "<NSLayoutConstraint:0x61800008dcf0 UIImageView:0x7fa012016350.width == UIView:0x7fa012024640.width   (active)>"
)

Will attempt to recover by breaking constraint 
<NSLayoutConstraint:0x618000096170 H:[UILabel:0x7fa01202bbf0'Grase Kendal']-(11)-|   (active, names: '|':UIView:0x7fa012024640 )>

Make a symbolic breakpoint at UIViewAlertForUnsatisfiableConstraints to catch this in the debugger.
The methods in the UIConstraintBasedLayoutDebugging category on UIView listed in <UIKit/UIView.h> may also be helpful.
2017-01-23 02:38:10.859468 DisplaySwitcher_Example[9356:1465466] [LayoutConstraints] Unable to simultaneously satisfy constraints.
	Probably at least one of the constraints in the following list is one you don't want. 
	Try this: 
		(1) look at each constraint and try to figure out which you don't expect; 
		(2) find the code that added the unwanted constraint or constraints and fix it. 
(
    "<NSLayoutConstraint:0x6100000946e0 H:|-(10)-[UILabel:0x7fa00ff206b0'Marry Kennedy']   (active, names: '|':UIView:0x7fa00ff20510 )>",
    "<NSLayoutConstraint:0x610000095220 H:[UILabel:0x7fa00ff206b0'Marry Kennedy']-(11)-|   (active, names: '|':UIView:0x7fa00ff20510 )>",
    "<NSLayoutConstraint:0x61000008cb20 UIImageView:0x7fa00ff20940.width == 1   (active)>",
    "<NSLayoutConstraint:0x6100000927f0 UIImageView:0x7fa00ff20940.width == UIView:0x7fa00ff20510.width   (active)>"
)

Will attempt to recover by breaking constraint 
<NSLayoutConstraint:0x610000095220 H:[UILabel:0x7fa00ff206b0'Marry Kennedy']-(11)-|   (active, names: '|':UIView:0x7fa00ff20510 )>

Make a symbolic breakpoint at UIViewAlertForUnsatisfiableConstraints to catch this in the debugger.
The methods in the UIConstraintBasedLayoutDebugging category on UIView listed in <UIKit/UIView.h> may also be helpful.
2017-01-23 02:38:10.860552 DisplaySwitcher_Example[9356:1465466] [LayoutConstraints] Unable to simultaneously satisfy constraints.
	Probably at least one of the constraints in the following list is one you don't want. 
	Try this: 
		(1) look at each constraint and try to figure out which you don't expect; 
		(2) find the code that added the unwanted constraint or constraints and fix it. 
(
    "<NSLayoutConstraint:0x6100000946e0 H:|-(10)-[UILabel:0x7fa00ff206b0'Marry Kennedy']   (active, names: '|':UIView:0x7fa00ff20510 )>",
    "<NSLayoutConstraint:0x610000095220 H:[UILabel:0x7fa00ff206b0'Marry Kennedy']-(11)-|   (active, names: '|':UIView:0x7fa00ff20510 )>",
    "<NSLayoutConstraint:0x61000008cb20 UIImageView:0x7fa00ff20940.width == 1   (active)>",
    "<NSLayoutConstraint:0x6100000927f0 UIImageView:0x7fa00ff20940.width == UIView:0x7fa00ff20510.width   (active)>"
)

Will attempt to recover by breaking constraint 
<NSLayoutConstraint:0x610000095220 H:[UILabel:0x7fa00ff206b0'Marry Kennedy']-(11)-|   (active, names: '|':UIView:0x7fa00ff20510 )>

Make a symbolic breakpoint at UIViewAlertForUnsatisfiableConstraints to catch this in the debugger.
The methods in the UIConstraintBasedLayoutDebugging category on UIView listed in <UIKit/UIView.h> may also be helpful.

**Comment by zhaxylykbayev** [comment]
same error
occurs when list is empty (while searching)

**Comment by freaqed** [comment]
closed via #27
